src/OpenTelemetry.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="MassTransitDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.MassTransit.Implementation
+{
+    internal class MassTransitDiagnosticListener : ListenerHandler
+    {
+        private readonly ActivitySourceAdapter activitySource;
+        private readonly MassTransitInstrumentationOptions options;
+
+        public MassTransitDiagnosticListener(ActivitySourceAdapter activitySource, MassTransitInstrumentationOptions options)
+            : base("MassTransit")
+        {
+            this.activitySource = activitySource;
+            this.options = options;
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (!this.options.TracedOperations.Contains(activity.OperationName))
+            {
+                return;
+            }
+
+            this.activitySource.Start(activity);
+            if (activity.IsAllDataRequested)
+            {
+                activity.DisplayName = this.GetDisplayName(activity);
```

Comment:
@dbarbashov Hey I just noticed you aren't calling `activity.SetKind` here, just wanted to confirm you didn't do that on purpose. Our other adapters I think all set Kind and DisplayName, and do so before calling the sampler (`activitySource.Start`). You might want to do something similar in case anyone relies on that?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3aca1299670fb9c3cb49b0cf1890818a6afc87ce/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs#L91-L94

Reply:
Yeah, I think we should always set display name & kind and set additional information if requested / available.

Reply:
Yeah I just didn't notice `SetKind` calls in other instrumentations. Thanks, will do!

Reply:
Fixed, thanks!